torch/_inductor/codegen/cpp.py

```diff
@@ -3799,10 +3783,27 @@ def _get_outer_loop_fusion_depth(self, node1, node2):
                 return outer_loop_fusion_depth
         return DISABLE_OUTER_LOOP_FUSION
 
+    def is_vertical_outer_loop_fusion(self, node1, node2):
+        return (
+            node1.get_names() & node2.ancestors
+            and not (
+                self._can_fuse_horizontal_impl(node1, node2)
+                and not node1.is_reduction()
```

Comment:
Just to confirm, this also includes scans, right (albeit they are not currently supported in cpp)

Reply:
I am not sure about the `scans` thing. Could you give an example? I think here is just to ensure it's not a norm fusion. Since norm fusion should be a super set of `OuterLoopFusion`.

Reply:
Scans are things like `torch.cumsum`. In this case, since scans also have to act on the whole tensor to get the result, you cannot fuse into them either (at least this way), same as with reductions really.

Reply:
Ok, thanks for the explanation. Looks like `torch.cumsum` has fallback to aten in CPU. 
Here we want to define `can_fuse_vertical_outer_loop` as:

- Node1 is an ancestors of Node2.
- Node1 and Node2 can't do norm fusion.
- Node1 and Node2 can fuse at some outer loop dimension.